internal/parser/types.go

```diff
@@ -25,5 +25,17 @@ type (
 		Comments []string
 	}
 
-	Type string
+	Type struct {
+
+		// Name of the type from outside its package
+		// Example `uuid.UUID` or `time.Time`
+		// Use this variable if you generate code outside the package of that type
+		// Do not forget to add corresponding imports in your template
+		Name string
+
+		// Name of the type from inside its package
+		// Example `UUID` or `Time`
+		// Use this variable if you generate code inside the package of that type
+		InternalName string
+	}
```

Comment:
IMO, it makes more sense to express Type as this. Then if you want to generate code outside of the package, you'll have to use: `{{.PackageName}}.{{Name}}`
```suggestion
	Type struct {

		// Name of the package the type belongs to
		// Example `uuid` or `time`
		// If you generate code outside this package do not forget to add corresponding imports in your template.
		// If needed
		PackageName string

		// Name of the type.
		// Example `UUID` or `Time`
	        Name string
	}
```

Reply:
Ok but how to you deal with types like 
`map[UUID]UUID` becoming `map[uuid.UUID]uuid.UUID` if expressed outside from the uuid package ?

Reply:
(also think about even more complex types, with structs.. and remember that the main use case is ''i want to get express the type being inside the PKG and outside, then I think two strings can cover 99% of our needs for now, we can enrich the struct later)

Reply:
It seems to be a problem outside of the scope of GenZ, let's use template helper or template functions to address that.

Reply:
I mean, let's keep type as it's expressed in the package seems to be sufficient enough at the moment

Reply:
> I mean, let's keep type as it's expressed in the package seems to be sufficient enough at the moment 

It's not sufficient at the moment, I want to generate some code types that are referenced from outside their package, for example google/uuid

Reply:
As discussed, this enhancement will come in future PR. Let's keep the current approach for the moment.